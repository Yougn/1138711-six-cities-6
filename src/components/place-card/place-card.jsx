import React from 'react';
import PropTypes from 'prop-types';
import {useHistory} from 'react-router-dom';
import {propCard} from '../common/propTypes';

const PlaceCard = (props) => {

  const history = useHistory();

  const {offer, mouseEnterCardId} = props;
  const {id, img, isFavorite, isPremium, price, rating, title, type} = offer;

  const mouseEnterIdHandler = ()=>{
    mouseEnterCardId(id);
  };

  return (
    <article className="cities__place-card place-card" onMouseEnter ={mouseEnterIdHandler} id={id}>
      { isPremium ? <div className="place-card__mark"><span>Premium</span></div> : ``}
      <div className="cities__image-wrapper place-card__image-wrapper">
        <a href="#">
          <img className="place-card__image" src={img} width="260" height="200" alt="Place image" id={id} />
        </a>
      </div>
      <div className="place-card__info" >
        <div className="place-card__price-wrapper" >
          <div className="place-card__price" >
            <b className="place-card__price-value" id={id}>&euro;{price}</b>
            <span className="place-card__price-text" id={id}>&#47;&nbsp;night</span>
          </div>
          <button className="place-card__bookmark-button button" type="button" id={id} onClick={() => history.push(`/favorites`)}>
            { isFavorite ?
              <svg className="place-card__bookmark-icon" width="18" height="19"><use xlinkHref="#icon-bookmark"></use></svg> :
              <svg className="place-card__bookmark-icon" width="18" height="19"><use xlinkHref="#icon-bookmark"></use></svg>}
            <span className="visually-hidden">To bookmarks</span>
          </button>
        </div>
        <div className="place-card__rating rating">
          <div className="place-card__stars rating__stars">
            <span style={{width: `80%`}}></span>
            <span className="visually-hidden" id={id}>{rating}</span>
          </div>
        </div>
        <h2 className="place-card__name" onClick={() => history.push(`/offer/:id`)}>
          <a href="#" id={id}>{title}</a>
        </h2>
        <p className="place-card__type" id={id}>{type}</p>
      </div>
    </article>
  );
};

PlaceCard.propTypes = {
  offer: PropTypes.shape(propCard),
  mouseEnterCardId: PropTypes.func.isRequired,
};

export default PlaceCard;
