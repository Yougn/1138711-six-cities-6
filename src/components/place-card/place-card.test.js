import React from 'react';
import {render} from '@testing-library/react';
import {PlaceCard} from './place-card';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';

test(`Should PlaceCard render correctly`, () => {
  const offer = {
    id: 1,
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
    price: 169,
    rating: 3,
    title: `The house among olive`,
    type: `room`,
    isPremium: true
  };
  const history = createMemoryHistory();

  const {container} = render(
      <Router history={history}>
        <PlaceCard offer={offer} />
      </Router>);
  expect(container).toMatchSnapshot();
});

