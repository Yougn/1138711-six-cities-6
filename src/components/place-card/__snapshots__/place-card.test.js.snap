// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should PlaceCard render correctly 1`] = `
<div>
  <article
    class="cities__place-card place-card"
    id="1"
  >
    <div
      class="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      class="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          class="place-card__image"
          height="200"
          src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      class="place-card__info"
    >
      <div
        class="place-card__price-wrapper"
      >
        <div
          class="place-card__price"
        >
          <b
            class="place-card__price-value"
          >
            €
            169
          </b>
          <span
            class="place-card__price-text"
          >
            / night
          </span>
        </div>
        <a
          href="/login"
        >
          <button
            class="place-card__bookmark-button button"
            type="button"
          >
            <svg
              class="place-card__bookmark-icon"
              height="19"
              width="18"
            >
              <use
                xlink:href="#icon-bookmark"
              />
            </svg>
             :
            <span
              class="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </a>
      </div>
      <div
        class="place-card__rating rating"
      >
        <div
          class="place-card__stars rating__stars"
        >
          <span
            style="width: 60%;"
          />
          <span
            class="visually-hidden"
          >
            3
          </span>
        </div>
      </div>
      <h2
        class="place-card__name"
      >
        <a
          href="/offer/1"
        >
          The house among olive
        </a>
      </h2>
      <p
        class="place-card__type"
      >
        room
      </p>
    </div>
  </article>
</div>
`;
